java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -170,6 +175,22 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
       assert offsetBuffer == null : "offsets are only supported for STRING";
     }
 
+    long[] children = new long[] {};
+    DataType dataType = new DataType(type);
```

Comment:
I would almost rather have us throw an exception if you try to use this constructor with a decimal type, and deprecate this constructor.  It could cause a lot of subtle bugs if we try to use it with a Decimal type and the scale is lost.